presto-main/src/main/java/com/facebook/presto/memory/MemoryPool.java

```diff
@@ -87,14 +104,17 @@ public void removeListener(MemoryPoolListener listener)
     /**
      * Reserves the given number of bytes. Caller should wait on the returned future, before allocating more memory.
      */
-    public ListenableFuture<?> reserve(QueryId queryId, long bytes)
+    public ListenableFuture<?> reserve(QueryId queryId, @Nullable String allocationTag, long bytes)
```

Comment:
> I'd prefer Optional instead of @Nullable
I don't have a strong opinion for `Nullable` vs. `Optional`, so I am fine updating the API to use `Optional`.

> why not just require all allocations be tagged?

There are calls that end up calling this method where we don't even have an allocation tag. For example, `QueryContext.setMemoryPool()`, or calls coming from `ChildAggregatedMemoryContext/RootAggregatedMemoryContext.closeContext()`, etc.